utilities/transactions/write_prepared_txn_db.cc

```diff
@@ -166,6 +166,16 @@ Status WritePreparedTxnDB::WriteInternal(const WriteOptions& write_options_orig,
     // increased for this batch.
     return Status::OK();
   }
+
+  if (batch->GetProtectionBytesPerKey() == 0 &&
```

Comment:
WriteBatch API protection_bytes_per_key | WriteOptions API protection_bytes_per_key | Final protection_bytes_per_key
-- | -- | --
0 | 0 | 0
0 | 8 | 8
8 | 0 | 8
8 | 8 | 8

Because of the third row. If user explicitly called a WriteBatch API to set protection_bytes_per_key to eight and provided a default WriteOptions (recall the default value of WriteOptions::protection_bytes_per_key is zero), we would want protection_bytes_per_key to remain eight. Explicit settings generally should override  implicit defaults.

Note the treatment is different in `TransactionBaseImpl::Reinitialize()` -- in that case the `WriteBatch` is being recycled thus any protection_bytes_per_key it has was not explicitly set for the upcoming operation. So in that case I allowed WriteOptions to overwrite to zero.